code/game/machinery/mind_machine.dm

```diff
@@ -0,0 +1,846 @@
+		errorMessage = "Safety consumed"
+	currentlySwapping = FALSE
+	occupantScan = FALSE
+	connectOne.icon_state = "mind_pod_closed"
+	connectTwo.icon_state = "mind_pod_closed"
+	icon_state = "mind_hub"
+	swapProgress = 0
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/theFlySwap(var/mob/living/MO, var/mob/living/MT)
+	var/list/mindFly = list()
+	var/list/simpFly = list()
+	MO = occupantOne
+	MT = occupantTwo
+	for(var/obj/item/weapon/holder/D in theFly)
+		for(var/mob/M in D.contents)
+			theFly += M //Gets the mob stored in the holder, adds it to the list
+		D.Destroy() //Turns them from holder to mob
+	theFly += MO
+	theFly += MT
+	for(var/mob/living/q in theFly)
+		if(q.mind)
+			mindFly += q
+			theFly -= q
+		else
+			simpFly += q
+			theFly -=q
+	var/mob/living/dummy = new(src.loc)
+	simpFly += dummy
+	var/mob/nuYou = null
+	for(var/mob/living/flies in mindFly)
+		nuYou = pick(simpFly)
+		if(flies.mind.special_verbs.len)
+			for(var/V in flies.mind.special_verbs)
+				flies.verbs -= V
+		for(var/spell/S in flies.spell_list)
+			flies.remove_spell(S)
+		flies.mind.transfer_to(nuYou)
+		simpFly += flies
+		simpFly -= nuYou
+	if(dummy.mind)
+		nuYou = pick(simpFly)
+		dummy.mind.transfer_to(nuYou)
+	qdel(dummy)
+	currentlySwapping = FALSE
+	occupantScan = FALSE
+	theFly.len = 0
+	simpFly.len = 0
+	mindFly.len = 0
+	connectOne.icon_state = "mind_pod_closed"
+	connectTwo.icon_state = "mind_pod_closed"
+	icon_state = "mind_hub"
+	badSwap = 1
+	swapProgress = 0
+	errorMessage = "Mind transport anomaly"
+
+//Swaps over///////////////
+//Pod stuff//////////
+
+/obj/machinery/mind_machine/mind_machine_pod
+	name = "Mind Machine Pod"
+	icon_state = "mind_pod_open"
+	desc = "A large pod used for mind transfers. Contains two locking systems: One for ensuring occupants do not disturb the transfer process, and another that prevents lower minded creatures from leaving on their own."
+	density = 1
+	anchored = 1.0
+	var/mob/living/occupant = null
+	var/podNumber = 0
+	var/obj/machinery/mind_machine/mind_machine_hub/connectedHub
+
+/obj/machinery/mind_machine/mind_machine_pod/New()
+	component_parts = newlist(
+		/obj/item/weapon/circuitboard/mind_machine_pod,
+		/obj/item/weapon/stock_parts/scanning_module/adv/phasic,
+		/obj/item/weapon/stock_parts/scanning_module/adv/phasic,
+		/obj/item/weapon/stock_parts/subspace/treatment,
+		/obj/item/weapon/stock_parts/subspace/ansible,
+		/obj/item/weapon/stock_parts/subspace/amplifier,
+		/obj/item/weapon/stock_parts/subspace/crystal,
+		/obj/item/weapon/stock_parts/subspace/transmitter
+		)
+
+/obj/machinery/mind_machine/mind_machine_pod/RefreshParts()
+	var/scanCount = 0
+	for(var/obj/item/weapon/stock_parts/SC in component_parts)
+		if(istype(SC, /obj/item/weapon/stock_parts/scanning_module))
+			scanCount += SC.rating
+			switch(podNumber)
+				if(1)
+					connectedHub.scanRatingOne = scanCount
+				if(2)
+					connectedHub.scanRatingTwo = scanCount
+
+/obj/machinery/mind_machine/mind_machine_pod/Destroy()
```

Comment:
This needs to notify the pod properly to avoid runtimes (e.g. if the pod is destroyed while actively scanning).